src/main/java/io/jenkins/pluginhealth/scoring/service/PluginService.java

```diff
@@ -24,15 +24,17 @@
 
 package io.jenkins.pluginhealth.scoring.service;
 
+import java.util.List;
 import java.util.stream.Stream;
-import javax.transaction.Transactional;
 
 import io.jenkins.pluginhealth.scoring.model.Plugin;
 import io.jenkins.pluginhealth.scoring.repository.PluginRepository;
 
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
+@Transactional(readOnly = true)
```

Comment:
I read this article: https://vladmihalcea.com/spring-transactional-annotation/ where the author suggests:

> For transactional services, it’s good practice to set the readOnly attribute to true at the class level and override it on a per-method basis for the service methods that need to write to the database.

Keeping this in mind, I removed the @Transactional annotation from the `streamAll` method since as per my understanding, it is a read-only transaction.


I'll undo the change if you find this change wrong and/or unnecessary.